chain/jsonrpc/src/api/validator.rs

```diff
@@ -49,3 +55,73 @@ impl RpcFrom<GetValidatorInfoError> for RpcValidatorError {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::api::RpcRequest;
+    use near_jsonrpc_primitives::types::validator::RpcValidatorRequest;
+    use near_primitives::hash::CryptoHash;
+    use near_primitives::types::{BlockId, EpochId, EpochReference};
+
+    #[test]
+    fn test_serialize_validators_params_as_vec() {
+        let block_hash = CryptoHash::new();
+        let block_hash_string = CryptoHash::new().to_string();
+        let params = serde_json::json!([block_hash_string]);
+        let result = RpcValidatorRequest::parse(params);
+        assert!(result.is_ok());
+        let result_unwrap = result.unwrap();
+        let res_serialized = format!("{result_unwrap:?}");
+        let expected = RpcValidatorRequest {
+            epoch_reference: EpochReference::BlockId(BlockId::Hash(block_hash)),
+        };
+        let expected_serialized = format!("{expected:?}");
+        assert_eq!(res_serialized, expected_serialized);
```

Comment:
This works, but the code is much simpler if you define `Eq` and `PartialEq` for `EpochReference`. Then you can simply do `assert_eq!(result.unwrap(), RpcValidatorRequest { epoch_reference: EpochReference::BlockId(BlockId::Hash(block_hash)) };`

I think it's acceptable because all network messages define `Eq` and `PartialEq`. `RpcValidatorRequest` is almost a network message. :)